java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/CopyFullyQualifiedClassNameAction.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.netbeans.modules.refactoring.java.ui;
+
+import java.awt.Toolkit;
+import java.awt.datatransfer.ClipboardOwner;
+import java.awt.datatransfer.StringSelection;
+import java.util.ArrayList;
+import java.util.List;
+import org.netbeans.api.java.classpath.ClassPath;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionReferences;
+import org.openide.awt.ActionRegistration;
+import org.openide.filesystems.FileObject;
+import org.openide.nodes.Node;
+import org.openide.util.HelpCtx;
+import org.openide.util.NbBundle;
+import org.openide.util.actions.NodeAction;
+
+/**
+ * Action to copy fully qualified name of selected classes.
+ * 
+ * @author Benjamin Asbach
+ */
+@ActionID(id = "org.netbeans.modules.refactoring.java.java.ui.CopyFullyQualifiedClassNameAction", category = "Edit")
+@ActionRegistration(displayName = "Copy Fully Qualified Class Name", lazy = true)
+@ActionReferences( value = {
+    @ActionReference(path = "Editors/text/x-java/Popup", position = 3500),
+    @ActionReference(path = "Loaders/text/x-java/Actions", position = 610)
+})
+public class CopyFullyQualifiedClassNameAction extends NodeAction {
+
+    private static final ClipboardOwner NO_CLIPBOARD_OWNER = null;
+
+    private final String name;
+    
+    public CopyFullyQualifiedClassNameAction() {
+        this.name = NbBundle.getMessage(this.getClass(), "LBL_CopyFullyQualifiedClassNameAction");
+    }
+    
+    @Override
+    protected void performAction(Node[] activatedNodes) {
+        List<String> fullyQualifiedClassNames = new ArrayList<>(activatedNodes.length);
+
+        for (Node activatedNode : activatedNodes) {
```

Comment:
The action attaches to Java editor, however if the activated node represents an editor, it does not work with caret position; so in a file with inner classes, just the top-level class' FQN is copied, although the caret may point on an inner class.

Reply:
Yes. You're right. Something I did not pay attention to. I wonder if implementing the caret processing is that easy. Maybe we should remove that context entry from the code view for now and just stick with that project view context menu. As providing this functionality only on file class level might be confusing as well.

Reply:
Removed that popup menu for the editor for now as it's not that easy to implement.